object/relations/relations.go

```diff
@@ -126,3 +129,230 @@ func ListRelations(ctx context.Context, rels Relations, cnrID cid.ID, rootObjID
 
 	return chain, nil
 }
+
+// HeadExecutor describes methods to get object head.
+type HeadExecutor interface {
+	ObjectHead(ctx context.Context, containerID cid.ID, objectID oid.ID, prm client.PrmObjectHead) (*client.ResObjectHead, error)
+}
+
+// SearchExecutor describes methods to search objects.
+type SearchExecutor interface {
+	ObjectSearchInit(ctx context.Context, containerID cid.ID, prm client.PrmObjectSearch) (*client.ObjectListReader, error)
+}
+
+// Executor describes all methods required to find all siblings for object.
+type Executor interface {
+	HeadExecutor
+	SearchExecutor
+}
+
+// All returns all related phy objects for provided root object ID in split-chain order.
+// If linking object is found its id will be the last one.
+//
+// Result doesn't include root object ID itself.
+func All(ctx context.Context, executor Executor, cnrID cid.ID, rootObjID oid.ID, tokens Tokens) ([]oid.ID, error) {
+	return WithLinking(ctx, executor, cnrID, rootObjID, tokens, true)
+}
+
+// WithLinking returns all related phy objects for provided root object ID in split-chain order.
+//
+// Result doesn't include root object ID itself.
+func WithLinking(ctx context.Context, executor Executor, containerID cid.ID, rootObjectID oid.ID, tokens Tokens, includeLinking bool) ([]oid.ID, error) {
+	splitInfo, err := getSplitInfo(ctx, executor, containerID, rootObjectID, tokens)
+	if err != nil {
+		if errors.Is(err, ErrNoSplitInfo) {
```

Comment:
can we describe it? i mean docs should say what happens when an object is not big. a user may expect `nil, nil` OR `[]oid.ID{}, nil`, OR `nil, ErrNoSplitInfo`, etc

Reply:
Docs updated